### ISSUE TITLE
Drush bash completion shows hosts

### ISSUE THREAD

**Issue by nalipaz** [drush-ops/drush#2213]
I have been having a strange issue. I have an alias named `example` on a server with `example_db` in the hosts entries (and possibly elsewhere since it is the hostname for the linked db server), typing something such as:

```
drush @exa[tab key]
```

Fills in with:

```
drush @example_db
```

rather than the expected

```
drush @example
```

If I run `drush sa` I do not see any entry for `@example_db` anywhere. Attempting to use that alias also results in an error saying the alias doesn't exist. Any ideas? It might or might not be noteworthy that this is in a Docker container.

**Comment by nalipaz** [comment]
CentOS 7
Drush 8.1.2 installed from the phar.

**Comment by nalipaz** [comment]
I now commented out the hosts entry for `example_db` and the completion does not occur on hitting the tab key anymore. I guess the question is then how can we avoid the bash completion of hosts entries in the aliases? Also, I then noticed that typing:
```
drush @[tab key]
```
displays all aliases on the site, but typing:
```
drush @exa[tab key]
```
does not fill in the remainder of the `@example` alias. I am fairly sure this did work in the past, am I incorrect in my recollection?

**Comment by Lawri-van-Buel** [comment]
The behaviour your describing for using the host file to complete a '@' is part of the bash completions from centos itself. (so not the one from drush) this might also be interfering with the drush one,

**Comment by nalipaz** [comment]
@Lawri-van-Buel nice catch. Seems it may not be CentOS specific, just a bash default. I found [a way to turn off bash's host-name completion](http://www.fvue.nl/wiki/Bash:_Completion_of_word_containing_@). 
```
shopt -u hostcomplete && complete -F _ssh ssh
```

However, the `drush @alias` completion still does not work, or perhaps it isn't supposed to?

**Comment by nalipaz** [comment]
It seems to me, from reading, that the '@' sign may be a predefined special character in bash completion? If that is the case I don't see how bash completion of aliases could work unless there is an escape sequence that we could use and drush actually does do the alias completions which I am not sure of at the moment.

I may just drop my alias name into my hosts file so that I can get the completion. And perhaps the completion I experienced in the past was exactly that, my hosts file coincidentally had a host name that matched the drush alias I defined.

**Comment by nalipaz** [comment]
So yeah, just tested dropping in the alias name on the hosts entries for 127.0.0.1, which shouldn't really hurt anything unless I run into a situation where I want to name a linked host with that name, and it does work to give the illusion of a bash drush @alias completion. It is up to you guys as to whether you want to keep this open and explore the possibilities of fixing/implemented drush alias completion. Either way, I am fine with my solution and will just keep it in my head as a potential gotcha.

**Comment by weitzman** [comment]
drush site alias completion is working for me on OSX. FYI, The best way to
setup drush completion now is to run drush init.

**Comment by nalipaz** [comment]
Yes @weitzman, all bash completion works with the exception of the site aliases. And yes I did run `drush -y init` in my Dockerfile. It seems that in many Linux versions of bash, from my understanding, the '@' sign is not working correctly as I described in my initial post.

**Comment by weitzman** [comment]
ping @grugnog

**Comment by nalipaz** [comment]
Testing further today I found that a production CentOS server was working correctly. This seems isolated to docker maybe...
